.github/workflows/e2e.yml

```diff
@@ -25,6 +26,13 @@ jobs:
         with:
           submodules: recursive
 
+      - name: Checkout subnet-evm
+        uses: actions/checkout@v4
+        with:
+          repository: ava-labs/subnet-evm
+          path: subnet-evm
+          ref: ${{ env.SUBNET_EVM_VERSION }}
+
```

Comment:
I don't think we should be downloading the `subnet-evm` repo in every case - that makes it just more of a pain than having it as a submodule. We should use the prebuilt binaries such as `https://github.com/ava-labs/subnet-evm/releases/download/v0.5.9/subnet-evm_0.5.9_darwin_arm64.tar.gz`, but be able to default back to a side-by-side installation of the repo, much like we do with the relayer.

This will probably necessitate doing the same thing for `avalanchego` since we won't be able to use the install scripts from `subnet-evm` - we can basically copy that one, and modify it to download and install both `subnet-evm` and `avalanchgo`.